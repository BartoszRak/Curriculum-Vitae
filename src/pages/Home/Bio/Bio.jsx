import React from 'react'
import PropTypes from 'prop-types'
import { withStyles, Typography, Divider } from '@material-ui/core'

import styles from './Bio.style'

export function Bio({ classes }) {
  return (
    <div className={classes.root} id="bio">
      <Typography className={classes.title} variant="h1">
        Wait, wait... but who Am I?
      </Typography>
      <Typography className={classes.subtitle} variant="h2">
        Unbrainless human
      </Typography>
      <Divider className={classes.divider} />
      <Typography variant="body1">
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
      </Typography>
      <Divider className={classes.divider} />
      <Typography variant="body1">
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
      </Typography>
    </div>
  )
}

Bio.propTypes = {
  classes: PropTypes.object,
}

export default withStyles(styles)(Bio)
