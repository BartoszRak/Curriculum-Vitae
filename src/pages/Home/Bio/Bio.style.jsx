export default ({ palette, spacing }) => ({
  divider: {
    margin: `${spacing.unit * 2}px 0`,
  },
  root: {},
  subtitle: {
    marginTop: spacing.unit * 2,
  },
  title: {
    color: palette.text.primary,
    fontSize: '3rem',
    marginBottom: spacing.unit * 4,
  },
})
