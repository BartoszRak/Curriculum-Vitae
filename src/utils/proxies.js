/* eslint-disable */
export const createPropertyProxy = (obj = {}) => new Proxy(obj, {
  get: (target, name) => target[name] || `${name}Mock`,
})

export const PropertyProxy = createPropertyProxy()
