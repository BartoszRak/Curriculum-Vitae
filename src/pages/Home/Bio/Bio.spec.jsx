import React from 'react'
import { cleanup } from '@testing-library/react'
import { shallow } from 'enzyme'

import { Bio } from './Bio'
import { PropertyProxy } from '~utils/proxies'

jest.unmock('@material-ui/core')

let props

describe('component: Bio', () => {
  afterEach(cleanup)

  beforeEach(() => {
    props = {
      classes: PropertyProxy,
    }
  })
  describe('rendering', () => {
    test('render without crash', () => {
      const wrapper = shallow(<Bio {...props} />)

      expect(wrapper).toBeTruthy()
    })

    test('match snapshot', () => {
      const wrapper = shallow(<Bio {...props} />)

      expect(wrapper).toMatchSnapshot()
    })
  })
})
