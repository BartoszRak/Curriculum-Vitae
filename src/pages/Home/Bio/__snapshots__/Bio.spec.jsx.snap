// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`global: Bio should render as expected 1`] = `
<div
  id="bio"
>
  <TypographyMock
    variant="h1"
  >
    Wait, wait... but who Am I?
  </TypographyMock>
  <TypographyMock
    variant="h2"
  >
    Unbrainless human
  </TypographyMock>
  <DividerMock />
  <TypographyMock
    variant="body1"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
  </TypographyMock>
  <DividerMock />
  <TypographyMock
    variant="body1"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
  </TypographyMock>
</div>
`;
